hbase-client/src/main/java/org/apache/hadoop/hbase/client/Scan.java

```diff
@@ -447,7 +450,16 @@ public Scan withStopRow(byte[] stopRow, boolean inclusive) {
    * after this method will yield undefined results.</b></p>
    * @param rowPrefix the prefix all rows must start with. (Set <i>null</i> to remove the filter.)
    * @return this
+   * @deprecated since 3.0.0. The scan result might be unexpected in some cases.
+   *   e.g. startRow : "112" and rowPrefixFilter : "11"
+   *   The Result of this scan might contains : "111"
+   *   This method implements the filter by setting startRow and stopRow,
+   *   but does not take care of the scene where startRow has been set.
```

Comment:
nit: `scene` can be replaced by `scenario`

Reply:
Updated.